src/Event.hh

```diff
@@ -2,53 +2,90 @@
 #define EVENT_H
 
 #include <string>
-#include <node_api.h>
-#include "wasm/include.h"
-#include <napi.h>
 #include <mutex>
 #include <map>
 
+#include <node_api.h>
+#include <napi.h>
+
+#include "wasm/include.h"
+
 using namespace Napi;
 
 struct Event {
   std::string path;
+  std::string pathTo;
+  std::string pathFrom;
   bool isCreated;
   bool isDeleted;
-  Event(std::string path) : path(path), isCreated(false), isDeleted(false) {}
+  bool isMoved;
+
+  Event(const std::string &path)
+      : path(path)
+      , isCreated(false)
+      , isDeleted(false)
+      , isMoved(false) {
+  }
 
-  Value toJS(const Env& env) {
+  Value toJS(const Env &env) {
     EscapableHandleScope scope(env);
     Object res = Object::New(env);
-    std::string type = isCreated ? "create" : isDeleted ? "delete" : "update";
     res.Set(String::New(env, "path"), String::New(env, path.c_str()));
-    res.Set(String::New(env, "type"), String::New(env, type.c_str()));
+    res.Set(String::New(env, "pathFrom"), String::New(env, pathFrom.c_str()));
+    res.Set(String::New(env, "pathTo"), String::New(env, pathTo.c_str()));
+    res.Set(String::New(env, "type"), String::New(env, (isMoved ? "move" : isCreated ? "create" : isDeleted ? "delete" : "update")));
```

Comment:
This would need to be done in a major version, with parity across all watcher backends. We need to ensure that the events emitted are consistent across platforms so that consumers can rely on the behavior.